src/StatsTools/variational_energy_estimator.jl

```diff
@@ -50,31 +51,31 @@ function variational_energy_estimator(shifts, overlaps; kwargs...)
     return ratio_of_means(numerator, denominator; kwargs...)
 end
 
-function variational_energy_estimator(sim; max_replicas=:all, kwargs...)
+function variational_energy_estimator(sim; max_replicas=:all, spectral_state=1, kwargs...)
     df = DataFrame(sim)
-    num_replicas = length(filter(startswith("norm_"), names(df))) # number of replicas
-    if iszero(num_replicas)
+    num_replicas = parse(Int, metadata(df, "num_replicas"))
+    if num_replicas == 1
         throw(ArgumentError(
             "No replicas found. Use keyword \
             `replica_strategy=AllOverlaps(n)` with n≥2 in `ProjectorMonteCarloProblem` to set up replicas!"
         ))
     end
     @assert num_replicas ≥ 2 "At least two replicas are needed, found $num_replicas"
 
-    num_overlaps = length(filter(startswith(r"c._dot"), names(df)))
+    num_overlaps = length(filter(startswith(Regex("r[0-9]+s$(spectral_state)_dot_r[0-9]+s$(spectral_state)")), names(df)))
```

Comment:
Does `Regex(r"...")` work? Shouldn't it be just
```suggestion
    num_overlaps = length(filter(startswith("r[0-9]+s$(spectral_state)_dot_r[0-9]+s$(spectral_state)"), names(df)))
```

Reply:
I think it has to be done with `Regex("...")` when you have interpolation (the `"r"` is just part of the string).

Reply:
Matija probably meant this way of writing it:
```suggestion
    num_overlaps = length(filter(startswith(r"r[0-9]+s$(spectral_state)_dot_r[0-9]+s$(spectral_state)"), names(df)))
```
I'm not sure whether this would make it more readable (if it works, I haven't tried it), so happy to leave it as it is.

Reply:
Actually, I think I'd prefer to store the number of overlaps in the DataFrame and retrieve it with a function  `num_overlaps()` similar to `num_replicas()` and `num_spectral_states`.

Reply:
Oh, I misread that! I think it needs to be done this way if you want to do string interpolation, so it's all good!